app/controllers/abilities_controller.rb

```diff
@@ -1,17 +1,30 @@
 class AbilitiesController < ApplicationController
+  before_action :set_user
+
   def index
-    @abilities = TrustLevel.all
+    @abilities = Ability.all
   end
 
   def show
-    @ability = TrustLevel.where(internal_id: params[:id]).first
+    @ability = Ability.where(internal_id: params[:id]).first
     return not_found if @ability.nil?
 
-    @your_ability = current_user.community_user.privilege @ability.internal_id
+    @your_ability = @user.community_user.privilege @ability.internal_id
   end
 
   def recalc
-    current_user.community_user.recalc_privileges
-    redirect_to abilities_url
+    @user.community_user.recalc_privileges
+    redirect_to abilities_url(for: @user.id)
+  end
+
+  private
+
+  def set_user
+    return not_found if current_user.nil?
+
+    @user = current_user
+    if params[:for]
```

Comment:
Deliberatly open for all. Nothing sensitive will be shown and it might be helpful in some cases (ex. support on Meta). For example, SE allows similar stuff, too.